starknet_py/contract.py

```diff
@@ -112,6 +114,69 @@ def __post_init__(self):
 InvocationResult = InvokeResult
 
 
+@add_sync_methods
+@dataclass(frozen=True)
+class DeclareResult(SentTransaction):
+    class_hash: int = None  # pyright: ignore
+    account: AccountClient = None  # pyright: ignore
+    compiled_contract: str = None  # pyright: ignore
+
+    def __post_init__(self):
+        assert self.class_hash is not None
+        assert self.account is not None
+        assert self.compiled_contract is not None
```

Comment:
Should we be using asserts in production code?

Reply:
Don't know, but `DeployResult` has it 
![image](https://user-images.githubusercontent.com/61014013/202208612-92c93f4d-60f2-4ead-8d04-4eb9060b7ab8.png)

Reply:
Perhaps we should change it here anyway, or maybe even in both places and explicitly do `if ... is not None: raise ValueError`? Seems like a cheap fix.

Reply:
I guess marshmallow dataclasses would be OK here as well, to validate those fields.

Reply:
Why can't we make those fields required, without default values = None?

Reply:
We've kind of decided agains using marshmallow dataclasses as the package is not very well maintained. We could use marshmallow schemas here, but it seems kind of overkill to me.